<!DOCTYPE html>
{% load static %}

<html lang="en" class="h-100" style="scroll-behavior:smooth;">

<head>

    {% include "viewer/dependencies/meta.html" %}
    {% include "viewer/dependencies/bootstrap.html" %}
    <link href="{% static "css/custom.css" %}" rel="stylesheet">
    <link href="{% static "css/sidebar.css" %}" rel="stylesheet">

    <script src="https://kit.fontawesome.com/988e732354.js" crossorigin="anonymous"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.8.3/modernizr.js"></script>
    <script type="text/javascript" src="https://npmcdn.com/tether@1.2.4/dist/js/tether.min.js"></script>
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"
            integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg=="
            crossorigin="anonymous">
    </script>
    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.min.js"
            integrity="sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN"
            crossorigin="anonymous">
    </script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js"
            integrity="sha384-B4gt1jrGC7Jh4AgTPSdUtOBvfO8shuf57BaghqFfPlYxofvL8/KUEfYiJOMMV+rV"
            crossorigin="anonymous">
    </script>

    <script type="text/javascript"
            src="https://cdnjs.cloudflare.com/ajax/libs/malihu-custom-scrollbar-plugin/3.1.5/jquery.mCustomScrollbar.concat.min.js"></script>
    <!-- Popper.JS -->
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.0/umd/popper.min.js"
            integrity="sha384-cs/chFZiN24E4KMATLdqdvsezGxaGsi4hLGOzlXwp5UZB1LY//20VyM2taTB4QvJ"
            crossorigin="anonymous"></script>
    <!-- Bootstrap JS -->
    <script type="text/javascript" src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/js/bootstrap.min.js"
            integrity="sha384-uefMccjFJAIv6A+rW+L4AHf99KvxDjWSu1z9VI8SKNVmz4sk7buKt/6v9KI65qnm"
            crossorigin="anonymous"></script>
    <!-- jQuery Custom Scroller CDN -->
    <script type="text/javascript"
            src="https://cdnjs.cloudflare.com/ajax/libs/malihu-custom-scrollbar-plugin/3.1.5/jquery.mCustomScrollbar.concat.min.js"></script>

    <!-- Scrollbar Custom CSS -->
    <link rel="stylesheet"
          href="https://cdnjs.cloudflare.com/ajax/libs/malihu-custom-scrollbar-plugin/3.1.5/jquery.mCustomScrollbar.min.css">
</head>

<body class="d-flex flex-column h-100">
<header class="mb-auto">
    {% include "viewer/includes/nav.html" %}
    {% include 'viewer/includes/messages.html' %}
</header>
<main role="main" class="flex-grow-1">

    <div class="wrapper">
        <!-- Sidebar -->
        <nav id="sidebar">
            <ul class="list-unstyled components">
                <h2 style="padding: 10px;">User Guide</h2>
                <li class="active">
                    <a href="#homeSubmenu" data-toggle="collapse" aria-expanded="false" class="dropdown-toggle">
                        <i class="fas fa-info-circle info-icon" style="font-size:22px; color:#999"></i>&nbsp; About
                        DecoPath
                    </a>
                    <ul class="collapse list-unstyled" id="homeSubmenu">
                        <li>
                            <a href="#background">Background</a>
                        </li>
                        <li>
                            <a href="#rightTool">Is DecoPath the right tool for me?</a>
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="#workflow"><i style="font-size:22px; color:#999" class="fa">&#xf15c;</i> &nbsp; DecoPath
                        workflow</a>
                </li>
                <li>
                    <a href="#testSubmenu" data-toggle="collapse" aria-expanded="false" class="dropdown-toggle">
                        <i style="font-size:22px; color:#999;" class="fa">&#xf058;</i>

                        &nbsp; Test run</a>
                    <ul class="collapse list-unstyled" id="testSubmenu">
                        <li>
                            <a href="#testORA">ORA test run</a>
                        </li>
                        <li>
                            <a href="#testGSEA">GSEA test run</a>
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="#pageSubmenu" data-toggle="collapse" aria-expanded="false"
                       class="dropdown-toggle collapsed">
                        <i style="font-size:22px; color:#999" class="fa">&#xf013;</i>
                        &nbsp; Running an experiment
                    </a>
                    <ul class="collapse list-unstyled" id="pageSubmenu">
                        <li>
                            <a href="#overview">Overview</a>
                        </li>
                        <li>
                            <a href="#upload">Uploading enrichment results</a>
                        </li>
                        <li>
                            <a href="#run">Running enrichment analysis</a>
                        </li>
                        <li>
                            <a href="#foldChanges">Differential gene expression analysis</a>
                        </li>
                        <li>
                            <a href="#experiments">Experiments page</a>
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="#resultsSubmenu" data-toggle="collapse" aria-expanded="false" class="dropdown-toggle">
                        <img style="width:25px;height:25px;" src="{% static 'icons/telescope.svg' %}">
                        &nbsp; Exploring results
                    </a>
                    <ul class="collapse list-unstyled" id="resultsSubmenu">
                        <li>
                            <a href="#ranking">Results page</a>
                        </li>
                        <li>
                            <a href="#consensus">Consensus table</a>
                        </li>
                        <li>
                            <a href="#hierarchy">Hierarchical view</a>
                        </li>
                        <li>
                            <a href="#overlap">Gene-level analysis</a>
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="#accountSubmenu" data-toggle="collapse" class="dropdown-toggle">
                        <i class="fas fa-user" style="font-size:22px; color:#999"></i>
                        &nbsp; Managing your account
                    </a>
                    <ul class="collapse list-unstyled" id="accountSubmenu">
                        <li>
                            <a href="#deleteExperiment">Delete experiment</a>
                        </li>
                        <li>
                            <a href="#deleteAccount">Delete account</a>
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="#contact"> <i style="font-size:22px; color:#999" class="fa">&#xf0e0;</i>&nbsp; Contact
                        us</a>
                </li>
            </ul>
        </nav>

        <!-- Page Content -->
        <div class="content">
            <section class="tutorial-section" id="background">
                <h2><strong>About DecoPath</strong></h2>
                <hr id="tutorialHeading">
                <h4>Background</h4>
                <hr>
                <p>
                    One of the more conventional methods to guide the interpretation of data generated by high
                    throughput (HT) technologies is determining if
                    sets of differentially expressed genes might be associated with a given biological process or a
                    phenotype. This can commonly be done through a pathway analysis, such as gene set enrichment
                    analysis (GSEA) or over representation analysis (ORA). While pathway analysis has certainly become
                    a highly popular approach, pathway boundaries can be loosely defined across pathway databases and
                    the selection of a particular database can come with an important caveat: <i>the choice of database
                    can impact the results of various pathway analysis.</i>
                </p>
                <p>
                    To shed light on what exactly a researcher can expect when they use a particular database for a
                    pathway analysis over another, we’ve developed DecoPath, a web application that can be used to
                    identify where
                    differences lie by comparing the results generated when pathways from different resources are used.
                </p>
                <p style="color:#212224">
                    <strong>DecoPath is an intuitive and extensible web application which allows users to easily
                        identify where these differences lie and which pathways are you can identify where these
                        differences lie by directly comparing results generated when pathways from different resources
                        are used.</strong>
                </p>
            </section>

            <section class="tutorial-section" id="rightTool">
                <h4>Is DecoPath the right tool for me?</h4>
                <hr>
                <p>The experimental design of a pathway analysis is a crucial first step in ensuring the results of an
                    analysis are meaningful. DecoPath is the right tool for you to:
                <ul>
                    <li>
                        Intuitively interpret the results of pathway analysis yielded by some of the most of the
                        well-known open-sourced pathway databases
                    </li>
                    <li>
                        Conduct or upload results of pathway analysis with variable experimental design implementations
                        to
                        easily identify inconsistencies across results
                    </li>
                </ul>
            </section>

            <section class="tutorial-section" id="workflow">
                <br>
                <h2><strong>DecoPath workflow</strong></h2>
                <hr id="tutorialHeading">
                <p id="workflowCaption">A typical DecoPath workflow consists of setting the configuration of an
                    experiment, uploading the required files to run it and finally, navigating to the experiments
                    page to explore the results.
                </p>
                <p>
                    By default, DecoPath includes 4 major pathway databases for pathway
                    gene sets (see
                    <a style="color: #007bff;" href={% url "imprint" %}>imprint</a>):
                </p>
                <ul>
                    <li>
                        <a style="color: #007bff;" href="https://www.genome.jp/kegg/pathway.html"
                           target="_blank">KEGG</a>
                    </li>
                    <li>
                        <a style="color: #007bff;" href="https://pathbank.org/" target="_blank">PathBank</a>
                    </li>
                    <li><a style="color: #007bff;" href="https://reactome.org/" target="_blank">Reactome</a>
                    </li>
                    <li>
                        <a style="color: #007bff;" href="https://www.wikipathways.org/index.php/WikiPathways"
                           target="_blank">WikiPathways</a>
                    </li>
                </ul>
                <p>
                    Details on each step of the workflow can be found below.
                </p>
                <div class="row" style="margin-bottom: -40px">
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/home.png' %}" style="width:100%;height:90%;">
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/ora_workflow.png' %}" style="width:100%;height:90%;">
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/gsea_circles.png' %}" style="width:100%;height:90%;">
                    </div>
                </div>
                <div class="row">
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">DecoPath home page</p>
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">Run experiments</p>
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">Explore results</p>
                    </div>
                </div>
            </section>

            <section class="tutorial-section" id="test">
                <section class="tutorial-section" id="testORA">
                    <br>
                    <h2><strong>Test run</strong></h2>
                    <hr id="tutorialHeading">
                    <h4>ORA test run</h4>
                    <hr>
                    <p>You can view and explore the results of running ORA on multiple databases for a liver cancer
                        dataset by downloading the file,
                        <a style="color: #007bff;"
                           href="https://drive.google.com/file/d/1f670R3Dv5HDMK9jlE4me0fY0E8SoBdwV/view?usp=sharing">
                            ORA results</a>.
                        Then, select ORA as the enrichment method in <strong>Step 2</strong>
                        of the <strong>Upload Results</strong> section of the
                        <a style="color: #007bff;" href="{% url 'home' %}">home page</a>
                        and upload this file.
                    </p>
                    <p>
                        You can also download the fold changes for genes from the same liver cancer dataset <a
                            style="color: #007bff;"
                            href="https://drive.google.com/file/d/1DyUS-tRgJTr_4po_2y-YduEnjhssRPqX/view?usp=sharing">
                        here</a> for an an additional, gene-level
                        analysis. Then, in <strong>Step 3</strong>, select "Upload differential gene expression analysis
                        results" and upload this file.
                    </p>
                    <p>
                        Click submit and you can then view the results on the Experiments page.
                    </p>
                </section>

                <section class="tutorial-section" id="testGSEA">
                    <h4>GSEA test run</h4>
                    <hr>
                    <p>You can view and explore the results of running GSEA on multiple databases for a liver cancer
                        dataset by downloading the file,
                        <a style="color: #007bff;"
                           href="https://drive.google.com/file/d/1QwvpqFofEv-Rfnb7BgfLe_RsKcrNiDF1/view?usp=sharing">
                            GSEA results</a>.
                        Then, select GSEA as the enrichment method in <strong>Step 2</strong>
                        of the <strong>Upload Results</strong> section of the
                        <a style="color: #007bff;" href="{% url 'home' %}">home page</a>
                        and upload this file.
                    </p>
                    <p>
                        You can also download the fold changes for genes from the same liver cancer dataset <a
                            style="color: #007bff;"
                            href="https://drive.google.com/file/d/1DyUS-tRgJTr_4po_2y-YduEnjhssRPqX/view?usp=sharing">
                        here</a> for an an additional, gene-level
                        analysis. Then, in <strong>Step 3</strong>, select "Upload differential gene expression analysis
                        results" and upload this file.
                    </p>
                    <p>
                        Click submit and you can then view the results on the Experiments page.
                    </p>
                </section>

            </section>

            <section class="tutorial-section" id="overview">
                <h2><strong>Running an experiment</strong></h2>
                <hr id="tutorialHeading">
                <h4>Overview</h4>
                <hr>
                <p>Users have 2 options to run an experiment:</p>
                <ol>
                    <li>
                        Upload the results of a pathway enrichment analysis
                    </li>
                    <li>
                        Conduct a pathway enrichment analysis using DecoPath
                    </li>
                </ol>
                <div class="row" style="margin-bottom: 5px">
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/upload.png' %}" style="width:100%;height:100%;">
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/ora_submit.png' %}" style="width:100%;height:100%;">
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <img src="{% static 'img/gsea_submit.png' %}" style="width:100%;height:100%;">
                    </div>
                </div>
                <div class="row">
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">Upload results</p>
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">Run ORA</p>
                    </div>
                    <div class="column" style="width: 33.3%;">
                        <p id="workflowSubCaption">Run GSEA</p>
                    </div>
                </div>
            </section>

            <section class="tutorial-section" id="upload">
                <br>
                <h4>Upload enrichment results</h4>
                <hr>
                <p>You can submit the results of an enrichment analysis and use the DecoPath visualizations and
                    functionalities to compare and explore the consensus and discrepancies around different pathway
                    databases. This is recommended for large pathway databases (e.g., with over 500 pathways) which can
                    be computationally expensive to run on the server. If you do
                    opt to submit your own results, you can download the gene sets from the
                    <a style="color: #007bff;" href="{% url 'home' %}">home page</a> and run GSEA or ORA by your
                    preferred method or by cloning the branch on the DecoPath
                    <a style="color: #007bff;"
                       href="https://github.com/DecoPath/DecoPath/tree/increase_processes">github page</a> and following
                    the instructions on the README.
                </p>
                <p>Details on how to prepare the files for upload can be found in the
                    <a style="color: #007bff;" href="{% url 'faqs' %}">FAQs</a>.
                </p>
            </section>

            <section class="tutorial-section" id="run">
                <h4>Run enrichment analysis</h4>
                <hr>
                <p>
                    For users opting to run analyses using DecoPath, the workflow depends on the analysis you select.
                    Users can select which of the four pathway databases they would like to include
                    in the analysis. By default, genesets from DecoPath which contain merged equivalent pathways are
                    also included in the analysis.
                </p>
                <p>
                    Detailed information on what types of data you can submit and information on how you can prepare the
                    files for upload can be found in the <a style="color: #007bff;" href="{% url 'faqs' %}">FAQs</a>.
                </p>
                <p>
                    We recommend downloading experiment results from the results page so analyses do not have to be run
                    again and you can upload files and use DecoPath functionalities at any time.
                </p>
                <p><b>Over Representation Analysis</b></p>
                <p>Running ORA requires:</p>
                <ul>
                    <li>A gene list</li>
                </ul>

                <p><b>Gene Set Enrichment Analysis</b></p>
                <p>To run GSEA, you will need:</p>
                <ul>
                    <li>An expression dataset</li>
                    <li>A file containing class labels (e.g., normal and tumor) for samples in the expression
                        dataset.
                    </li>
                </ul>
                <p>
                    Users also have the option to run GSEA on a pre-ranked list of genes. For GSEA pre-ranked, users
                    must upload a file containing a ranked list of genes, similar to an
                    <a style="color: #007bff;"
                       href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#Ranked_Gene_Lists">
                        RNK file</a> (Ranked list file).
                </p>
            </section>

            <section class="tutorial-section" id="foldChanges">
                <h4>Differential gene expression analysis</h4>
                <hr>
                <p>
                    While you can generate visualizations without differential gene expression (DGE)
                    analysis, by conducting this supplementary analysis, you can also analyze pathways at
                    the gene level. This allows users to identify which genes are responsible for any disparate results
                    that may be observed across databases.
                </p>
                <p>Further details on this analysis can be found in the
                    <a style="color: #007bff;" href="{% url 'faqs' %}">FAQs</a>.
                </p>
                <p>
                    We recommend downloading DGE analysis results from the Gene-Level Analysis page so analyses do not
                    have to be run again and you can upload files and use DecoPath functionalities at any time.
                </p>
            </section>

            <section class="tutorial-section" id="experiments">
                <h4>Experiments page</h4>
                <hr>
                <p>
                    Once you have submitted an experiment, you can find it on the
                    <a style="color: #007bff;" href="{% url 'experiments' %}">Experiments</a> page along with details on
                    whether it has run successfully, is pending or has failed. You can refresh the page for the latest
                    status.
                </p>
                <p>
                    If the experiment has failed, you can click on the <strong style="color:red;">X</strong> icon in the
                    "Status" column to see what the issue may have been.
                </p>
                <p>Note the expected run times for an analysis:</p>
                <ul>
                    <li>ORA: 0-10 minutes</li>
                    <li>GSEA: 10-120 minutes</li>
                    <li>DGE analysis: 5-30 minutes</li>
                </ul>
            </section>

            <section class="tutorial-section" id="ranking">
                <br>
                <h2><strong>Exploring results</strong></h2>
                <hr id="tutorialHeading">
                <h4>Results page</h4>
                <hr>
                <p>
                    This page summarizes the results of the pathway analysis performed. For ORA, this includes
                    details on the pathway and the <i>q-</i>value denoting whether the pathway is enriched in the gene
                    list. For GSEA, the report also includes the (normalized) enrichment scores for each pathway.
                </p>
            </section>

            <section class="tutorial-section" id="consensus">
                <h4>Consensus table</h4>
                <hr>
                <p>
                    The Consensus page visualization shows the consensus of the results of enrichment analysis across
                    databases at the pathway level.
                </p>
                <p>
                    For GSEA, the table displays the normalized enrichment score (NES)
                    for a given pathway across each database as well as the NES of the merged gene sets of all
                    equivalent pathways in the column “DecoPath”. Pathways that are not significantly enriched according
                    to a user-defined cut-off are coloured gray. Similarly for ORA, the table shows the <i>q</i>-value
                    for a given pathway across each database and pathways that are not significantly enriched are
                    coloured gray.
                </p>
                <img class="centerImage" src="{% static 'img/consensus.png' %}" style="width:50%;height:50%;">
            </section>

            <section class="tutorial-section" id="hierarchy">
                <br>
                <h4>Hierarchical view</h4>
                <hr>
                <p>
                    The hierachical view is an interactive visualization of equivalent pathways across databases. The
                    organization of pathways into major categories allows users to intuitively navigate through
                    the hierarchy and identify the consensus (or discrepancies) around enriched pathways. These
                    pathways are categorized into 8 major categories:
                </p>
                <ul>
                    <li>Metabolism</li>
                    <li>Immune system</li>
                    <li>Signaling</li>
                    <li>Communication and transport</li>
                    <li>Programmed cell-death</li>
                    <li>Disease pathways</li>
                    <li>DNA repair and replication</li>
                    <li>Others</li>
                </ul>
                <p>
                    In total, the hierarchy contains 644 pathways and can be found
                    <a href="https://github.com/ComPath/compath-resources/blob/master/mappings/decopath_ontology.xlsx"
                       style="color: #007bff;"> here</a>.
                </p>
                <div class="row">
                    <div class="column" style="width: 50%; margin-bottom:-50px;">
                        <img src="{% static 'img/circles_ora.png' %}" style="width:100%;height:90%;">
                    </div>
                    <div class="column" style="width: 50%;">
                        <img src="{% static 'img/gsea_circles.png' %}" style="width:100%;height:95%;">
                    </div>
                </div>
                <div class="row">
                    <div class="column" style="width: 50%;">
                        <p id="workflowSubCaption">ORA results hierarchical view</p>
                    </div>
                    <div class="column" style="width: 50%;">
                        <p id="workflowSubCaption">GSEA results hierarchical view</p>
                    </div>
                </div>
            </section>

            <section class="tutorial-section" id="overlap">
                <h4>Gene-level analysis</h4>
                <p>
                    This interactive visualization shows the overlap of genes for equivalent pathways to pinpoint
                    specific genes that may contribute to the contradictions observed in the results of
                    enrichment analysis. If you have uploaded fold changes of differentially expressed genes or run
                    DGE analysis, you can also view the distribution of fold changes of genes in your dataset.
                </p>
                <p>
                    In the images below, we zoom-in on the results of GSEA using KEGG, WikiPathways and Reactome for a
                    liver cancer dataset. We focus on the database-specific results for the hedgehog signaling pathway.
                </p>
                <p>
                    With a supplementary DGE analysis, you can plot the distribution of fold changes for
                    differentially expressed genes in your dataset to identify which genes are
                    leading to either consistent or contradictory results of the pathway analysis.
                </p>

                <div class="row">
                    <div class="column" style="width: 34%;">
                        <img src="{% static 'img/hedgehog.png' %}" style="width:100%;height:100%;">
                    </div>
                    <div class="column" style="width: 33%;">
                        <img src="{% static 'img/hedgehog_venn.png' %}" style="width:100%;height:100%;">
                    </div>
                    <div class="column" style="width: 33%;">
                        <img src="{% static 'img/hedgehog_dist.png' %}" style="width:100%;height:100%;">
                    </div>
                </div>

                <div class="row">
                    <div class="column" style="width: 34%;">
                        <p id="workflowSubCaption"
                           style="text-align: justify; padding-left: 20px; padding-right: 10px; font-size: 16px">
                            Example GSEA results in which the hedgehog signaling pathway is not significantly enriched
                            using KEGG and WikiPathways (gray circles) but is significantly enriched using Reactome
                            (blue circle).
                        </p>
                    </div>
                    <div class="column" style="width: 33%;">
                        <p id="workflowSubCaption"
                           style="text-align: justify; padding-left: 5px; padding-right: 5px; font-size: 16px">
                            The Venn diagram depicts the overlap of genes for the hedgehog signaling pathway. Gene sets
                            from KEGG and WikiPathways heavily overlap with each
                            other, but minimally with the pathway from Reactome, explaining why Reactome presents with
                            discordant results.
                        </p>
                    </div>
                    <div class="column" style="width: 33%;">
                        <p id="workflowSubCaption"
                           style="text-align: justify; padding-left: 10px; padding-right: 20px; font-size: 16px">
                            Plotting the distribution of fold changes for each of the genes in the gene sets illustrates
                            how discordant results arise.
                        </p>
                    </div>
                </div>


            </section>

            <section class="tutorial-section" id="deleteExperiment">
                <br>
                <h2><strong>Managing your account</strong></h2>
                <hr id="tutorialHeading">
                <h4>Delete experiment</h4>
                <p>
                    It's easy to delete an experiment! Just head over to the
                    <a style="color: #007bff;" href={% url 'experiments' %}>Experiments</a>
                    page, hover over the experiment you want to delete, and click on the trash can icon.
                </p>
                <img class="centerImage" src="{% static 'img/delete_experiment.png' %}" style="width:50%;height:50%;">
            </section>

            <section class="tutorial-section" id="deleteAccount">
                <br>
                <h4>Delete account</h4>
                <p>Deleting your account is also simple. You can delete your account at anytime on the <a
                        style="color: #007bff;" href={% url 'account' %}>Account</a> page.</p>
                <img class="centerImage" src="{% static 'img/delete_account.png' %}" style="width:50%;height:50%;">
            </section>

            <section class="tutorial-section" id="contact">
                <h2><strong>Contact us</strong></h2>
                <p>
                    DecoPath is currently developed and maintained in an academic capacity by Sarah Mubeen, Vinay
                    Srinivas Bharadhwaj and Daniel Domingo-Fernández at the
                    <a style="color: #007bff;"
                       href="https://www.scai.fraunhofer.de/en/business-research-areas/bioinformatics.html">
                        Fraunhofer SCAI Department of Bioinformatics
                    </a>
                </p>
                <p>
                    Write us an email, report a bug or send us a feature request at sarah.mubeen@scai.fraunhofer.de.
                    You can also report a bug by creating an issue on our
                    <a href="https://github.com/DecoPath" style="color: #007bff;">github</a> page.
                </p>
            </section>

            <div class="footer-basic">
                <footer class="footer bg-light">
                    <div style="text-align: center; padding-bottom:25px;">
                        <a href="https://biorxiv.org/" target="_blank">
                            <i class="fa fa-file-text" style="font-size:36px;color:#2e7c97"></i>
                            <a href="https://github.com/" target="_blank">
                                <i class="fa fa-github" style="font-size:36px; color:#2e7c97;"></i>
                            </a>
                            <br>
                        </a>
                    </div>
                    <ul class="list-inline">
                        <li class="list-inline-item"><a href="{% url "home" %}">Home</a></li>
                        <li class="list-inline-item"><a href="#">Contact Us</a></li>
                        <li class="list-inline-item"><a href="{% url "imprint" %}">Terms and Conditions</a></li>
                        <li class="list-inline-item"><a href="{% url "data_protection" %}">Privacy Policy</a></li>
                    </ul>
                    <p class="copyright">DecoPath © 2021 Fraunhofer SCAI</p>
                </footer>
            </div>
        </div>
    </div>
</main>


<script type="text/javascript">

    $(document).ready(function () {

        $("#sidebar").mCustomScrollbar({
            theme: "minimal"
        });

        $('#sidebarCollapse').on('click', function () {
            // open or close navbar
            $('#sidebar').toggleClass('active');
            // close dropdowns
            $('.collapse.in').toggleClass('in');
            // and also adjust aria-expanded attributes we use for the open/closed arrows
            // in our CSS
            $('a[aria-expanded=true]').attr('aria-expanded', 'false');
        });

    });

    // When the user scrolls the page, execute myFunction
    window.onscroll = function () {
        myFunction()
    };

    // Get the navbar
    var navbar = document.getElementById("navbar");

    // Get the offset position of the navbar
    var sticky = navbar.offsetTop;

    // Add the sticky class to the navbar when you reach its scroll position. Remove "sticky" when you leave the scroll position
    function myFunction() {
        if (window.pageYOffset >= sticky) {
            navbar.classList.add("sticky")
        } else {
            navbar.classList.remove("sticky");
        }
    }

    $(window).scroll(function () {

        var height = $('.wrapper').height();
        $('#sidebar').height(height)

        var dist = $(window).scrollTop();
        var sTop = $('#sidebar').position().top;
        var mHeight = $('.wrapper').height();
        var userDist = 2000;
        if ((sTop > (mHeight - userDist)) && (dist > (mHeight - userDist))) {
            //the sidebar is pinned now. it won't scroll further.
        } else if (dist < (mHeight - userDist)) {
            $('#sidebar').animate({
                top: dist + $('#sidebar').height()
            }, 0);
        }
    });

</script>


